Composer/packages/client/src/CreationFlow/DefineConversation/index.js

```diff
@@ -34,33 +32,11 @@ const validateForm = data => {
 export function DefineConversation(props) {
   const { state } = useContext(StoreContext);
   const { onSubmit, onGetErrorMessage, onDismiss, enableLocationBrowse } = props;
-  const { templateId, focusedStorageFolder } = state;
-  const [formData, setFormData] = useState({ errors: {} });
+  const { templateId } = state;
+  const [formData, setFormData] = useState({ name: templateId, errors: {} });
```

Comment:
This doesn't append with a number if a bot already exists with this name right? Isn't that what we want?

Reply:
I change the code back to the last version. Just like my last comment, our code structure is like
<A>
 update name based on location
 <B/>
</A>
<B>
  will update location
</B>
We need to wait B finish updating location then the number in name can be calculated. For this feature, a better code structure should be after B getting the location variable, then passing it to A as a prop. 
<B>
will update location
 <A location/>
</B>
<A>
   update name based on location
</A>
But for this small feature we do not need to refactor the code.

So use an useEffect(() => set default name, [location]) is necessary. And recalculation on name becomes a side effect. What do you think?

Reply:
I see 2 options, both remove the need for the `defaultName` state variable:

1. if `templateId` and `focusedStorageFolder` are set on the first render, we do not need the `useEffect` hook.
2. If they are not set, we can use the `useEffect` hook with a guard. 
```ts
useEffect(() => {
  if (!name) {
    // calculate name
   setFormData({ name: '...' });
  }
}, [templateId, focusedStorageFolder]);
```

Reply:
A wrong name will be set on the first render. Then focuedStorageFolder will be updated. But the name cannot be updated due to if(!name). I think I need to refactor code. Update on focusedStorageFolder should be put on the upper layer rather than locationSelectContent component just like what Dong Lei said.